### PR TITLE
(PA-3114) Use facter-ng on all platform

### PR DESCRIPTION
**Pull request by mihaibuzgau**

### CODE REVIEW

**Comment by oanatmaria** [comment]
Tests passed: https://jenkins-master-prod-1.delivery.puppetlabs.net/view/Adhoc/job/platform_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/749/

File: acceptance/tests/ensure_puppet_facts_works_with_facter_ng.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  teardown do
+    on agent, puppet('config set facterng false')
+  end
+
+  agents.each do |agent|
+    step "test puppet facts with facter-ng'" do
+      on agent, puppet('config set facterng true')
+      on agent, puppet('facts'), :acceptable_exit_codes => [0] do
+        facter_major_version = stdout.chomp.split("\n").select{|line|  line.match?(/facterversion/)}
+        assert_match(/4.[0-9]+.[0-9]+/, facter_major_version.first, "puppet failed to change cfacter to facter-ng")
```
Comment (gimmyxd) [inline acceptance/tests/ensure_puppet_facts_works_with_facter_ng.rb]:
```
facter_major_version = stdout.split("\n").find { |line|  line.match?(/facterversion/) }
assert_match(/4.[0-9]+.[0-9]+/, facter_major_version, "puppet failed to change cfacter to facter-ng")
```

`find` will stop at the first occurrence. Alternatively you could `JSON.prase(stdout)` and access the `facterversion` key

File: configs/components/facter-ng.rb [hunk 2]
```diff
@@ -17,8 +17,13 @@
     pkg.environment "RUBYLIB", "#{settings[:ruby_vendordir]}:$(RUBYLIB)"
   end
 
+  sed_pattern = %(s/spec.required_ruby_version = '~> 2.3'/ /)
```
Comment (gimmyxd) [inline configs/components/facter-ng.rb]: Should we remove this only if platform is cross compiled?

**Comment by mihaibuzgau** [comment]
tests passed: https://jenkins-master-prod-1.delivery.puppetlabs.net/view/Adhoc/job/platform_puppet-agent-extra_puppet-agent-integration-suite_adhoc-ad_hoc/752/

**Pull request merged**